### ISSUE TITLE
Definition of U gate is not valid

### ISSUE THREAD

**Issue by invalidexplorer** [Qiskit/qiskit#1165]
Hi, I tried implementing U gate via the decomposition of it given on qiskit docs (https://qiskit.org/documentation/stubs/qiskit.circuit.library.UGate.html) . However the results from the decomposition and U gate are not concurrent.

u3(pi/2,pi/3,pi/4) q[0] gives:  [ 0.707+0j, 0.354+0.612j ]

while 
rz(pi/3) q[0];
rx(-pi/2) q[0];
rz(pi/2) q[0];
rx(pi/2) q[0];
rz(pi/4) q[0];

gives: [ 0.43-0.561j, -0.701+0.092j ]

Is it an issue or am I doing something wrong?

**Comment by navaneethsdk** [comment]
Yes, I guess you are right. I just run a simulation and got different results. But chances are that I may be wrong too. I would love to have the mistake pointed out (if there is one).
![issue-test](https://user-images.githubusercontent.com/56035039/107763333-7e628b80-6d54-11eb-9f4f-d3e397fa5f46.png)

**Comment by codegiri** [comment]
Did you account for the global phase factor into the implementation? I rapidly computed the unitary matrix for this sequence of gates on paper, I think you can easily check that RZ(phi-pi/2)*RX(pi/2)*RZ(pi-theta)*RX(pi/2)*RZ(lambda-pi/2) = exp(-j(phi+lambda)/2)*U3(theta,phi,lambda).
If you account for this phase factor in your example, which will be exp(-7pi/24*j), I believe you should get the desired result.
